subsys/bluetooth/host/hci_core.c

```diff
@@ -5060,53 +5090,63 @@ int bt_le_adv_start(const struct bt_le_adv_param *param,
 		return -EALREADY;
 	}
 
-	d[0].data = ad;
-	d[0].len = ad_len;
+	if (!(param->options & BT_LE_ADV_OPT_DIR_MODE_LOW_DUTY) &&
+	    !(param->options & BT_LE_ADV_OPT_DIR_MODE)) {
+		struct bt_ad d[2] = {};
 
-	err = set_ad(BT_HCI_OP_LE_SET_ADV_DATA, d, 1);
-	if (err) {
-		return err;
-	}
+		d[0].data = ad;
+		d[0].len = ad_len;
 
-	d[0].data = sd;
-	d[0].len = sd_len;
+		err = set_ad(BT_HCI_OP_LE_SET_ADV_DATA, d, 1);
+		if (err) {
+			return err;
+		}
 
-	if (param->options & BT_LE_ADV_OPT_USE_NAME) {
-		const char *name;
+		d[0].data = sd;
+		d[0].len = sd_len;
 
-		if (sd) {
-			int i;
+		if (param->options & BT_LE_ADV_OPT_USE_NAME) {
+			const char *name;
 
-			/* Cannot use name if name is already set */
-			for (i = 0; i < sd_len; i++) {
-				if (sd[i].type == BT_DATA_NAME_COMPLETE ||
-				    sd[i].type == BT_DATA_NAME_SHORTENED) {
-					return -EINVAL;
+			if (sd) {
+				int i;
+
+				/* Cannot use name if name is already set */
+				for (i = 0; i < sd_len; i++) {
+					if (sd[i].type
+						== BT_DATA_NAME_COMPLETE ||
+					    sd[i].type
+						== BT_DATA_NAME_SHORTENED) {
+						return -EINVAL;
```

Comment:
Ugh.. this deep indentation causes some really ugly line splits. Could you try to redesign this somehow, there are many possibilities like inverted conditions, labels, helper functions, etc).